contrib/exporters/vpclogs/mod/transform.go

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/contrib/exporters/core"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/logging"
+)
+
+// NewTransform creates a new flow transformer based on a name string
+func NewTransform(cfg *viper.Viper) (interface{}, error) {
+	excludeStartedFlows := cfg.GetBool(core.CfgRoot + "transform.secadvisor.exclude_started_flows")
+
+	return &vpclogsFlowTransformer{
+		flowUpdateCountCache: cache.New(10*time.Minute, 10*time.Minute),
+		excludeStartedFlows:  excludeStartedFlows,
+	}, nil
+}
+
+const (
+	VpcActionReject string = "R"
+	VpcActionAccept string = "A"
+)
+
+// VpclogsFlow represents a vpc log flow
+// this structure will be updated when the fields are finalized
+type VpclogsFlow struct {
+	FinishType          flow.FlowFinishType `json:"-"`
```

Comment:
why do these fields appear here if not serialized? please remove all fields annotated with hyphen.

Reply:
Some fields are stored to save intermediate information. These fields are later used to derive the fields that are in the output specification. Some fields may be used when computing the action Accet/Reject, and they need to be available at that time. They are kept as internal fields and not included in the final output.